kyaml/yaml/rnode.go

```diff
@@ -512,24 +526,57 @@ func (rn *RNode) SetAnnotations(m map[string]string) error {
 }
 
 // GetLabels gets the metadata labels field.
+// If specific labels are provided, only those are returned.
+// If no specific labels are provided, then all are returned.
 // If the field is missing, returns an empty map.
 // Use another method to check for missing metadata.
-func (rn *RNode) GetLabels() map[string]string {
+func (rn *RNode) GetLabels(labels ...string) map[string]string {
 	meta := rn.getMetaData()
 	if meta == nil {
 		return make(map[string]string)
 	}
-	return rn.getMapFromMeta(meta, LabelsField)
+	return rn.getMapFromMeta(meta, LabelsField, labels...)
 }
 
 // getMapFromMeta returns map, sometimes empty, from metadata.
-func (rn *RNode) getMapFromMeta(meta *RNode, fName string) map[string]string {
-	result := make(map[string]string)
-	if f := meta.Field(fName); !f.IsNilOrEmpty() {
-		_ = f.Value.VisitFields(func(node *MapNode) error {
-			result[GetValue(node.Key)] = GetValue(node.Value)
-			return nil
-		})
+func (rn *RNode) getMapFromMeta(meta *yaml.Node, fName string, fields ...string) map[string]string {
+	f := yNodeField(meta, fName)
+	if f == nil {
+		return make(map[string]string)
+	}
+	if f.Kind != MappingNode {
+		return nil
+	}
+	if len(fields) == 0 {
+		return fieldsAsStringMapAll(f)
+	}
+	return fieldsAsStringMapSome(f, fields...)
+}
+
+// fieldsAsStringMapAll returns a map of the node's content.
+func fieldsAsStringMapAll(n *yaml.Node) map[string]string {
+	result := make(map[string]string, len(n.Content))
+	for i := 0; i < len(n.Content); i = i + 2 {
+		key := n.Content[i].Value
+		value := n.Content[i+1].Value
+		result[key] = value
+	}
+	return result
+}
+
+// fieldsAsStringMapSome returns a map of the node's content,
+// restricted to the specified fields.
+func fieldsAsStringMapSome(n *yaml.Node, fields ...string) map[string]string {
+	size := len(fields)
+	result := make(map[string]string, size)
+	for i := 0; i < len(n.Content); i = i + 2 {
```

Comment:
```suggestion
	for i := 0; i < len(n.Content);  i = IncrementFieldIndex(i) {
```

I know it adds an extra layer, but `IncrementFieldIndex` keeps things encapsulated and makes the code easier to understand.

Reply:
If this code stands after I investigate, I'm happy to swap it. I saw both in the file and wasn't certain what was expected.